### ISSUE TITLE
Sampling via py scripts

### ISSUE THREAD

**Issue by eyalbetzalel** [EugenHotaj/pytorch-generative#16]
Hi!

Is there a way to use the sample function that appears in the notebook within the py scripts / add sampling option to the tensorboard logs? 

I tried to convert the imshow to imsave and run it after finish training, but the output image is currrpted for some reason. 

BTW - this project is great! thank you for this. 

Regards.

**Comment by EugenHotaj** [comment]
> Is there a way to use the sample function that appears in the notebook within the py scripts / add sampling option to the tensorboard logs?

All the implemented models have a [#sample()](https://github.com/EugenHotaj/pytorch-generative/blob/master/pytorch_generative/models/base.py#L35-L61) method which they inherit from the base class. Does this not work for you?

> I tried to convert the imshow to imsave and run it after finish training, but the output image is currrpted for some reason.

Hmm, interesting, let me look more into this and get back to you.

> BTW - this project is great! thank you for this.

Thanks! Glad you like it and I appreciate the bug reports.

**Comment by EugenHotaj** [comment]
Hey @eyalbetzalel 

> > I tried to convert the imshow to imsave and run it after finish training, but the output image is currrpted for some reason.
> 
> Hmm, interesting, let me look more into this and get back to you.

I built a quick notebook with the PixelSnail model ([colab link](https://colab.research.google.com/drive/1gW9tBTwV390CGx2Y_tJqfLYIK6WsDlUR?usp=sharing)). I only ran it for one epoch so the generated images are not very high quality, but as far as I can tell everything seems to be working.

How exactly are the output images corrupted for you? Are you not even able to open the saved image file? Or do the results just look bad?

> add sampling option to the tensorboard logs

Yea it should be possible to add images to TensorBoard using the [SummaryWriter API](https://pytorch.org/docs/master/tensorboard.html#torch.utils.tensorboard.writer.SummaryWriter.add_image). I've been hesitant to add this feature since generating samples takes a while, especially fo big models, which could slow down training. However, if we do it pretty infrequently (say every ~10 epochs) it should not be a big deal. I'll open a new issue to track this.
